### PR TITLE
OSD-8789 Add osde2e-stage-rosa-e2e-byo-vpc-proxy-postinstall job

### PR DESCRIPTION
**Pull request by mrbarge**
This PR:

- Sets the `osde2e-stage-rosa-e2e-byo-vpc-proxy-install` to no longer hibernate, as it otherwise gets re-used by other E2E jobs, which we don't want.

- Adds a new job for the cluster-wide proxy day 2 testing. This job requires/uses the same secrets as its sibling `osde2e-stage-rosa-e2e-byo-vpc-proxy-install` job but will not install a cluster with a proxy. It is configured to run on weekdays roughly two hours after the `osde2e-stage-rosa-e2e-byo-vpc-proxy-install` cluster should expire and uninstall.

Refs: [OSD-8789](https://issues.redhat.com//browse/OSD-8789) / [SDE-1634](https://issues.redhat.com//browse/SDE-1634)

### CODE REVIEW

**Comment by mrbarge** [comment]
@MrSantamaria would you be able to sanity-check this one please?

File: ci-operator/jobs/openshift/osde2e/openshift-osde2e-main-periodics.yaml [hunk 1]
```diff
@@ -3065,6 +3067,67 @@ periodics:
...
+- agent: kubernetes
+  cluster: build03
+  cron: 0 5 * * 1-5
+  decorate: true
+  extra_refs:
+  - base_ref: main
+    org: openshift
+    repo: osde2e
+  labels:
+    pj-rehearse.openshift.io/can-be-rehearsed: "false"
```
Comment (MrSantamaria) [inline ci-operator/jobs/openshift/osde2e/openshift-osde2e-main-periodics.yaml:3079]: We can enable this to true so it will run under the pj-rehearse

Comment (mrbarge) [reply ci-operator/jobs/openshift/osde2e/openshift-osde2e-main-periodics.yaml]: Ahhh, thanks! I've set that now.

**Comment by mrbarge** [comment]
@MrSantamaria could I ask for an lgtm off you for this one now please? :smiley:

**Pull request merged**